CorePayments/src/main/java/com/paypal/android/corepayments/UpdateClientConfigAPI.kt

```diff
@@ -0,0 +1,119 @@
+package com.paypal.android.corepayments
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+import com.paypal.android.corepayments.graphql.GraphQLClient
+import com.paypal.android.corepayments.graphql.GraphQLRequest
+import com.paypal.android.corepayments.graphql.GraphQLResult
+import kotlinx.serialization.InternalSerializationApi
+import kotlinx.serialization.Serializable
+
+@OptIn(InternalSerializationApi::class)
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class UpdateClientConfigAPI(
+    private val applicationContext: Context,
+    private val graphQLClient: GraphQLClient,
+    private val resourceLoader: ResourceLoader
+) {
+
+    constructor(context: Context, coreConfig: CoreConfig) : this(
+        context.applicationContext,
+        GraphQLClient(coreConfig),
+        ResourceLoader()
+    )
+
+    object Defaults {
+        const val INTEGRATION_ARTIFACT = "MOBILE_SDK"
+        const val USER_EXPERIENCE_FLOW = "INCONTEXT"
+        const val PRODUCT_FLOW = "HERMES"
+    }
+
+    suspend fun updateClientConfig(
+        orderId: String,
+        fundingSource: String
+    ): UpdateClientConfigResult {
+        return when (val result = resourceLoader.loadRawResource(
+            applicationContext,
+            R.raw.graphql_query_update_client_config
+        )) {
+            is LoadRawResourceResult.Success -> {
+                val graphQLRequest = createUpdateClientConfigRequest(
+                    orderId = orderId,
+                    fundingSource = fundingSource,
+                    query = result.value
+                )
+
+                val graphQLResponse =
+                    graphQLClient.send<UpdateClientConfigResponse, UpdateClientConfigVariables>(
+                        graphQLRequest = graphQLRequest
+                    )
+                when (graphQLResponse) {
+                    is GraphQLResult.Success -> {
+                        val correlationId = graphQLResponse.correlationId
+                        graphQLResponse.response.data?.let {
+                            UpdateClientConfigResult.Success
+                        } ?: UpdateClientConfigResult.Failure(
+                            APIClientError.noResponseData(
+                                correlationId
+                            )
```

Comment:
```suggestion
                            APIClientError.noResponseData(correlationId)
```
Does this break the line limit? Single line would be more readable here.